### PR TITLE
feat(ui) Create page for managing home page posts

### PR DESCRIPTION
**Pull request by chriscollins3456**
First off I want to give a HUGE shoutout to @PatrickfBraz for his work that went into this PR here. He did the vast majority of this in his PR here: https://github.com/datahub-project/datahub/pull/8292
His work is what enabled us to push this over the finish line otherwise who knows how long it would take before we got there. So thanks so much again!

This PR creates a new page in settings where users that have the platform privilege for managing home page posts can go to in order to create and delete home page posts of either text or link kind. Right now we don't allow editing since that resolver doesn't exist yet.

This also makes some styling updates to make posts look nicer.

**Screenshots:**
image

image

image

image

image

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/authorization/AuthorizationUtils.java [hunk 1]
```diff
@@ -107,7 +107,31 @@ public static boolean canEditGroupMembers(@Nonnull String groupUrnStr, @Nonnull
   }
 
   public static boolean canCreateGlobalAnnouncements(@Nonnull QueryContext context) {
-    return isAuthorized(context, Optional.empty(), PoliciesConfig.CREATE_GLOBAL_ANNOUNCEMENTS_PRIVILEGE);
+    final DisjunctivePrivilegeGroup orPrivilegeGroups = new DisjunctivePrivilegeGroup(
+        ImmutableList.of(
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/authorization/AuthorizationUtils.java:111]: nice!

File: datahub-web-react/src/app/search/PostLinkCard.tsx [hunk 2]
```diff
@@ -74,19 +79,21 @@ export const PostLinkCard = ({ linkPost }: Props) => {
     const link = linkPost?.content?.link || '';
 
     return (
-        <CardContainer type="link" href={link}>
+        <CardContainer type="link" href={link} target="_blank" rel="noopener noreferrer">
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/search/PostLinkCard.tsx:82]: thank you!! so much better than redirecting.

File: datahub-web-react/src/app/settings/SettingsPage.tsx [hunk 3]
```diff
@@ -143,6 +147,11 @@ export const SettingsPage = () => {
                                 <TeamOutlined /> <ItemTitle>Ownership Types</ItemTitle>
                             </Menu.Item>
                         )}
+                        {showHomePagePosts && (
+                            <Menu.Item key="posts">
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/settings/SettingsPage.tsx:151]: fire!

File: datahub-web-react/src/app/settings/posts/CreatePostForm.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                            name={LOCATION_FIELD_NAME}
+                            rules={[{ type: 'url', warningOnly: true }]}
+                            hasFeedback
+                        >
+                            <Input data-testid="create-post-media-location" placeholder="Your post image URL" />
+                        </SubFormItem>
+                    </SubFormItem>
+                </>
+            )}
+        </Form>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/settings/posts/CreatePostForm.tsx:89]: looks great. copy also improved!

thanks for adding the select with conditional fields

File: datahub-web-react/src/app/settings/posts/CreatePostModal.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+                        form.getFieldValue(DESCRIPTION_FIELD_NAME),
+                        form.getFieldValue(LINK_FIELD_NAME),
+                        form.getFieldValue(LOCATION_FIELD_NAME),
+                    );
+                    form.resetFields();
+                }
+            })
+            .catch((e) => {
+                message.destroy();
+                message.error({ content: `Failed to create Post!: \n ${e.message || ''}`, duration: 3 });
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/settings/posts/CreatePostModal.tsx:72]: minor: i've been trying to remove places where we print the raw backend message, and replace with a console.error + An unknown error occurred for the end user

Comment (jjoyce0510) [reply datahub-web-react/src/app/settings/posts/CreatePostModal.tsx:72]: (tiny)

File: datahub-web-react/src/app/settings/posts/CreatePostModal.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            });
+        onClose();
+    };
+
+    // Handle the Enter press
+    useEnterKeyListener({
+        querySelectorToExecuteClick: '#createPostButton',
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/settings/posts/CreatePostModal.tsx:79]: nice!!

File: datahub-web-react/src/app/settings/posts/ManagePosts.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+`;
+
+const ListContainer = styled.div``;
+
+export default function ManagePosts() {
+    return (
+        <PageContainer>
+            <PageHeaderContainer>
+                <PageTitle level={3}>Home Page Posts</PageTitle>
+                <Typography.Paragraph type="secondary">View and manage your DataHub Posts.</Typography.Paragraph>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/settings/posts/ManagePosts.tsx:31]: Maybe add something about the posts appearing on the landing page.

"View and manage the pinned posts that appear to users on the landing page. "

File: datahub-web-react/src/app/settings/posts/PostItemMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+type Props = {
+    urn: string;
+    title: string;
+    onDelete?: () => void;
+};
+
+export default function PostItemMenu({ title, urn, onDelete }: Props) {
+    const [deletePostMutation] = useDeletePostMutation();
+
+    const deleteDomain = () => {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/settings/posts/PostItemMenu.tsx:16]: nit: deletePost

File: datahub-web-react/src/app/settings/posts/PostsList.tsx [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+            title: 'Description',
+            dataIndex: '',
+            key: 'description',
+            render: (record: PostEntry) => PostColumn(record.description || ''),
+        },
+        {
+            title: 'Type',
+            dataIndex: '',
+            key: 'type',
+            render: (record: PostEntry) => PostColumn(record.contentType),
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/settings/posts/PostsList.tsx:101]: reminder: we should make sure the copy shown aligns with what the user created (Announcement vs Link)

File: datahub-web-react/src/app/settings/posts/constants.ts [hunk 10]
```diff
@@ -0,0 +1,6 @@
+export const TITLE_FIELD_NAME = 'title';
+export const DESCRIPTION_FIELD_NAME = 'description';
+export const LINK_FIELD_NAME = 'link';
+export const LOCATION_FIELD_NAME = 'location';
+export const TYPE_FIELD_NAME = 'switch';
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/settings/posts/constants.ts:5]: it's called "switch"? does type not work?

**Review by jjoyce0510 — APPROVED** [review]
Overall looks great. Minor comments only.

File: datahub-web-react/src/app/settings/posts/CreatePostModal.tsx (hunk 5 above)
Comment (chriscollins3456) [reply datahub-web-react/src/app/settings/posts/CreatePostModal.tsx:72]: hear that

File: datahub-web-react/src/app/settings/posts/ManagePosts.tsx (hunk 7 above)
Comment (chriscollins3456) [reply datahub-web-react/src/app/settings/posts/ManagePosts.tsx:31]: oh yeah so much better

File: datahub-web-react/src/app/settings/posts/PostItemMenu.tsx (hunk 8 above)
Comment (chriscollins3456) [reply datahub-web-react/src/app/settings/posts/PostItemMenu.tsx:16]: nicee

File: datahub-web-react/src/app/settings/posts/PostsList.tsx (hunk 9 above)
Comment (chriscollins3456) [reply datahub-web-react/src/app/settings/posts/PostsList.tsx:101]: definitely good call will update this

File: datahub-web-react/src/app/settings/posts/constants.ts (hunk 10 above)
Comment (chriscollins3456) [reply datahub-web-react/src/app/settings/posts/constants.ts:5]: it does lol it used to be called `SWITCH_FIELD_NAME` but i changed it and forgot to update the value here (even though technically it doesn't matter cuz it's just a const that's used) - but updating!

**Pull request merged**